src/core/producer/layer.cpp

```diff
@@ -107,6 +107,9 @@ struct layer::impl
             }
 
             state_ = foreground_->state();
+            state_["producer"]               = foreground_->name();
+            state_["background"] = background_->state();
+            state_["background"]["producer"] = background_->name();
```

Comment:
I would like to move foreground to /foreground so we have background & foreground. Would make more sense?

Reply:
We could do that, I didn't do this because it would be duplicate in some way. But it's an easy change so I'll do that if you want me to.

Reply:
Please change that. And move your other change into another PR. I'd like to merge this while I'm a little unsure about the other one.